src/Stack/Fetch.hs

```diff
@@ -531,7 +531,9 @@ fetchPackages' mdistDir toFetchAll = do
                 S.writeFile cabalFP $ tfCabal toFetch
 
                 atomically $ modifyTVar outputVar $ Map.insert ident destDir
-            $logWarn $ mconcat $ map (\(path, entryType) -> "Unexpected entry type " <> entryType <> " for entry " <> T.pack path) unexpectedEntries
+
+            F.forM_ unexpectedEntries $ \(path, entryType) ->
+                logWarnN $ "Unexpected entry type " <> entryType <> " for entry " <> T.pack path
```

Comment:
FYI, stack never uses `logWarnN`, since it doesn't record source file and position. Otherwise LGTM.